bundle-hack/update_csv.go

```diff
@@ -0,0 +1,416 @@
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	if err := updateManifestWithRedHatDetails(manifest); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	// Finally, write the updated manifest
+	if err := writeManifest(manifest, manifestDirectory, version); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	fmt.Printf("Successfully updated CSV manifest for %s.\n", version)
+}
+
+// getManifestFilePathFromDirectory scans the directory and returns the first file
+// ending with "clusterserviceversion.yaml".
+func getManifestFilePathFromDirectory(dir string) (string, error) {
+	files, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return "", err
+	}
+
+	for _, f := range files {
+		if strings.HasSuffix(f.Name(), "clusterserviceversion.yaml") {
+			return filepath.Join(dir, f.Name()), nil
+		}
+	}
+	return "", errors.New("clusterserviceversion.yaml not found")
+}
+
+// addRequiredAnnotations adds required annotations to the CSV file content represented as a map.
+func addRequiredAnnotations(m map[string]interface{}) error {
+	requiredAnnotations := map[string]string{
+		"features.operators.openshift.io/disconnected":      "true",
+		"features.operators.openshift.io/fips-compliant":    "true",
+		"features.operators.openshift.io/proxy-aware":       "false",
+		"features.operators.openshift.io/tls-profiles":      "false",
+		"features.operators.openshift.io/token-auth-aws":    "false",
+		"features.operators.openshift.io/token-auth-azure":  "false",
+		"features.operators.openshift.io/token-auth-gcp":    "false",
+		"operatorframework.io/suggested-namespace":          "openshift-security-profiles",
+	}
+
+	metadata, ok := m["metadata"].(map[string]interface{})
+	if !ok {
+		return errors.New("error: 'metadata' does not exist in the CSV content")
+	}
+
+	annotations, ok := metadata["annotations"].(map[string]interface{})
+	if !ok {
+		return errors.New("error: 'annotations' does not exist within 'metadata' in the CSV content")
+	}
+
+	for k, v := range requiredAnnotations {
+		annotations[k] = v
+	}
+
+	fmt.Println("Successfully added required annotations.")
+	return nil
+}
+
+// addRequiredLabels adds required labels under metadata.labels
+func addRequiredLabels(m map[string]interface{}) error {
+	requiredLabels := map[string]string{
+		"operatorframework.io/arch.amd64": "supported",
+	}
+
+	metadata, ok := m["metadata"].(map[string]interface{})
+	if !ok {
+		return errors.New("error: 'metadata' does not exist in the CSV content")
+	}
+
+	labels, found := metadata["labels"].(map[string]interface{})
+	if !found {
+		labels = make(map[string]interface{})
+		metadata["labels"] = labels
+	}
+
+	for k, v := range requiredLabels {
+		labels[k] = v
+	}
+
+	fmt.Println("Successfully added required labels.")
+	return nil
+}
+
+// replaceImages updates the operator/SELinuxD images.
+func replaceImages(m map[string]interface{}) error {
+	const (
+		SPO_OPERATOR_IMAGE_PULLSPEC       = "quay.io/redhat-user-workloads/ocp-isc-tenant/security-profiles-operator"
```

Comment:
I think these will need to point to a registry.redhat.io reference for the bundle image to be something we can ship, right?